openfl-tutorials/experimental/Workflow_Interface_401_FedProx_with_Synthetic_nonIID.ipynb

```diff
@@ -0,0 +1,802 @@
+    "                        ), \n",
+    "                        batch_size=batch_size, shuffle=True\n",
+    "                    )\n",
+    "            }"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that we have defined dataset class. Let define model, optimizer, and some helper functions like we would for any other deep learning experiment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from openfl.interface.aggregation_functions.weighted_average import weighted_average as wa\n",
+    "\n",
+    "\n",
+    "class Net(nn.Module):\n",
+    "    \"\"\"\n",
+    "    Model to train the dataset\n",
+    "\n",
+    "    Args:\n",
+    "        None\n",
+    "    \n",
+    "    Returns:\n",
+    "        model: class Net object\n",
+    "    \"\"\"\n",
+    "    def __init__(self):\n",
+    "        # Set RANDOM_STATE to reproduce same model\n",
+    "        pt.set_rng_state(pt.manual_seed(RANDOM_SEED).get_state())\n",
+    "        super(Net, self).__init__()\n",
+    "        self.linear1 = nn.Linear(60, 100)\n",
+    "        self.linear2 = nn.Linear(100, 10)\n",
+    "\n",
+    "    def forward(self, x):\n",
+    "        x = self.linear1(x)\n",
+    "        x = self.linear2(x)\n",
+    "        return x\n",
+    "\n",
+    "\n",
+    "def cross_entropy(output, target):\n",
+    "    \"\"\"\n",
+    "    cross-entropy metric\n",
+    "\n",
+    "    Args:\n",
+    "        output: model ouput,\n",
+    "        target: target label\n",
+    "\n",
+    "    Returns:\n",
+    "        crossentropy_loss: float\n",
+    "    \"\"\"\n",
+    "    return F.cross_entropy(output, pt.max(target, 1)[1])\n",
+    "\n",
+    "\n",
+    "def compute_loss_and_acc(network, dataloader):\n",
+    "    \"\"\"\n",
+    "    Model test method\n",
+    "\n",
+    "    Args:\n",
+    "        network: class Net object (model)\n",
+    "        dataloader: torch.utils.data.DataLoader\n",
+    "\n",
+    "    Returns:\n",
+    "        (accuracy,\n",
+    "        loss,\n",
+    "        correct,\n",
+    "        dataloader_size)\n",
+    "    \"\"\"\n",
+    "    network.eval()\n",
+    "    test_loss = 0\n",
+    "    correct = 0\n",
+    "    with pt.no_grad():\n",
+    "        for data, target in dataloader:\n",
+    "            output = network(data)\n",
+    "            test_loss += cross_entropy(output, target).item()\n",
+    "            tar = target.argmax(dim=1, keepdim=True)\n",
+    "            pred = output.argmax(dim=1, keepdim=True)\n",
+    "            correct += pred.eq(tar).sum().cpu().numpy()\n",
+    "    dataloader_size = len(dataloader.dataset)\n",
+    "    test_loss /= dataloader_size\n",
+    "    accuracy = float(correct / dataloader_size)\n",
+    "    return accuracy, test_loss, correct\n",
+    "\n",
+    "\n",
+    "def weighted_average(tensors, weights):\n",
```

Comment:
I propose that we create `openfl.experimental.interface.aggregation_functions` and move this function there.

Reply:
Incorporated